### PR TITLE
fix(DB/Creature): Adjust Lich King position during last phase of DK quest

### LINKED ISSUE #9175: DK starting Area bugged quest
### Current Behaviour

When i try to finish the quest The Scarlet Apocalypse, i have to talk to the Lichking in Deaths Breach. Well he is not on his Platform. Heßs  in the Platform. You can still turn the quest in but its a little bit buggy.

### Expected Blizzlike Behaviour

He should Stand atop the platform.

### Source

_No response_

### Steps to reproduce the problem

Start a Dk
make the storyquests till you get the quest The Scarlet Apocalypse
Try talking to the Lichking in Deats Breach.

### Extra Notes

_No response_

### AC rev. hash/commit

b2f0893

### Operating system

Win 10 running with docker

### Custom changes or Modules

_No response_

### PR DESCRIPTION
**Pull request by Voxstrasza**
## Changes Proposed:
-  Adjust Lich King position on platform so he consistently spawns on it
-  #9180 helped the problem but it wasn't consistent as his spawn coordinates clipped the model's feet into the platform, 50/50 chance of whether or not he'd stay or fall through onto the ground below

## Issues Addressed:
- Permanently closes #9175

## Tests Performed:
[After](https://i.imgur.com/dzxzwQm.jpg)

Feet are no longer clipped!

Before SQL, 5 server restarts yielded 2/5 times that LK stayed in position
After SQL, 5 server restarts yielded 5/5 times that LK stayed in position

## How to Test the Changes:
.go c 130896

### CODE REVIEW

**Comment by Nyeriah** [comment]
Hey,

Thanks for the contribution, we really appreciate your support. However, we should try to verify his spawn position through data mining to confirm this before doing any changes.

@acidmanifesto do you have anything on this one?

Regards

**Comment by TheLegendaryKingdom** [comment]
Tested it by applying the SQL query in your update, then restarted the server multiple times (shoutout to AC/TC booting time being only 3 seconds !) and after the third restart, lich king had his feet inside the ground but was still on top of the platform.
Same thing happenned around two more times during my restarts (not as deeply in-ground than the third time) but still not fell into the platform. I performed around 10-15 restarts, so I would say by so far it does not seem falling (again, if it depends on luck...) but still gets his feet inside the ground sometimes
WoWScrnShot_120421_205900
WoWScrnShot_120421_205145
.

**Comment by Voxstrasza** [comment]
> Hey,
> 
> Thanks for the contribution, we really appreciate your support. However, we should try to verify his spawn position through data mining to confirm this before doing any changes.
> 
> @acidmanifesto do you have anything on this one?
> 
> Regards



> Hey,
> 
> Thanks for the contribution, we really appreciate your support. However, we should try to verify his spawn position through data mining to confirm this before doing any changes.
> 
> @acidmanifesto do you have anything on this one?
> 
> Regards

Hey, all this does is move him up so his feet don't clip anymore.  If you have coordinates that don't clip, feel free to share.



> Tested it by applying the SQL query in your update, then restarted the server multiple times (shoutout to AC/TC booting time being only 3 seconds !) and after the third restart, lich king had his feet inside the ground but was _still on top_ of the platform. Same thing happenned _around two more times_ during my restarts but LK still not fell into the platform.
> 
> I performed around 10-15 restarts, so I would say by so far it does not seem falling (again, if it depends on luck...) but still gets his feet inside the ground sometimes.
> 
> WoWScrnShot_120421_205900 WoWScrnShot_120421_205145 .

I'm noticing you're using an updated graphical mod and have GM on.  Please try without both.  I'm not having the same issue.

**Comment by acidmanifesto** [comment]
I can parse my sniffs later with the dk zone

**Comment by TheLegendaryKingdom** [comment]
I increased just a little bit the position_z field : from 161.148 to 161.3 and after over a dozen of restarts it had by so far never merged into the ground. Putting him a bit upper seem to fix this (for now, with a dozen of restarts and not 3000 which cannot tell if luck still decides here). Maybe if sniffed packets reveal a lower position_z it will need to be different on AC than the sniffed position to correctly work here ? IDK (still unexperienced yet) if it is a bad thing to adjust position of a creature like that for the AC project.

File: data/sql/updates/pending_db_world/rev_1638646536763994700.sql
Comment (TheLegendaryKingdom) [inline data/sql/updates/pending_db_world/rev_1638646536763994700.sql:8]: Seems to work better with `position_z = 161.3`

Comment (Voxstrasza) [reply data/sql/updates/pending_db_world/rev_1638646536763994700.sql:8]: Thanks for this.  I agree.  I'll apply it.

Comment (Voxstrasza) [reply data/sql/updates/pending_db_world/rev_1638646536763994700.sql]: It's been applied.  Give it a few more tries if you are able.

Comment (TheLegendaryKingdom) [reply data/sql/updates/pending_db_world/rev_1638646536763994700.sql]: Still not merged his feet inside the ground after a dozen more restarts.

Btw the gm state should not be related, and my mod only affects weapons models (and fixed helmet geoset on this helmet) and should therefore not interact with this issue (I guess ? Still, I tried without it, and the result is the same than with. I'll eat then try by completing the quests instead of gm on if needed, but Idon't really think so tbh).

Comment (Voxstrasza) [reply data/sql/updates/pending_db_world/rev_1638646536763994700.sql]: I mention the GM mode because it usually creates wonky results and seems like a bug (when a normal player could come across the same thing, it wouldn't be a bug)

If your tests don't make a difference with GM on, that's a plus!

Comment (TheLegendaryKingdom) [reply data/sql/updates/pending_db_world/rev_1638646536763994700.sql]: Noticed. I'll try the regular "player" way after eating (:

**Comment by acidmanifesto** [comment]
give me a few minutes. im going ot parse and see what hte sniff position is.

**Comment by acidmanifesto** [comment]
Ok. so Parsing gives me the following info:

INSERT INTO `creature` (`guid`, `id`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `PhaseGroup`, `modelid`, `equipment_id`, `position_x`, `position_y`, `position_z`, `orientation`, `spawntimesecs`, `spawndist`, `currentwaypoint`, `curhealth`, `curmana`, `MovementType`, `npcflag`, `unit_flags`, `dynamicflags`, `VerifiedBuild`) VALUES
(@CGUID+2274, 29110, 609, 4298, 4342, 1, 1, 0, 0, 0, 2310.266845703125, -5742.21240234375, 161.207427978515625, 3.857177734375, 120, 0, 0, 3208000, 0, 0, 0, 0, 0, 41079); -- 29110 (Area: 4342 - Difficulty: 0) (Auras: 53274 - 53274)

so xyzo is :  2310.266845703125, -5742.21240234375, 161.207427978515625, 3.857177734375,

However due to us only recognizing 6 digits in hole. the xzyo will then be in return 
2310.26, -5742.21 161.207, 3.85717
If we do not do the rounding meathod.

If we were to do the rounding meathod then the xyzo will then be in return
2310.27, -5742.21, 161.207, 3.85718

**Comment by acidmanifesto** [comment]
@Nyeriah i submitted the sniff via the sniff team repo for your own personal comparison if you want.
But at this point, it is up for debate, although the parsing shows a bit of a slight adjustment, it isnt much but may be much depending on how the Azerothcore is wanting to accept whatever adjustments based on the sniff and parse.
@Voxstrasza @TheLegendaryKingdom That info is per official sniffing and parsing, i wrote the sniffing standard guides so will let you two debate from there, your current adjustments is not too far from official values, but since i provided the official values, please reconsider readjusting your values to as close as possible to the official values.

**Comment by Kitzunu** [comment]
Use the given values.

i even believe you can use the exact value and it will round itself.

**Comment by TheLegendaryKingdom** [comment]
> @Nyeriah i submitted the sniff via the sniff team repo for your own personal comparison if you want. But at this point, it is up for debate, although the parsing shows a bit of a slight adjustment, it isnt much but may be much depending on how the Azerothcore is wanting to accept whatever adjustments based on the sniff and parse. @Voxstrasza @TheLegendaryKingdom That info is per official sniffing and parsing, i wrote the sniffing standard guides so will let you two debate from there, your current adjustments is not too far from official values, but since i provided the official values, please reconsider readjusting your values to as close as possible to the official values.

I tested with the values your parsed, and **after the first restart LK's feet were into the ground**.
However, the other coordinates seem more accurate. So, I only edited position_z from the parse to be **161.21** and after a dozen more restarts it did **not** merge to the ground.

By the way, I used the raw values and like @Kitzunu said _it automatically rounded_ them.

I suggest using the **official values** and adjust only `position_z = 161.21`.

**Comment by Voxstrasza** [comment]
> Use the given values.
> 
> i even believe you can use the exact value and it will round itself.

It does round itself.



> @Nyeriah i submitted the sniff via the sniff team repo for your own personal comparison if you want. But at this point, it is up for debate, although the parsing shows a bit of a slight adjustment, it isnt much but may be much depending on how the Azerothcore is wanting to accept whatever adjustments based on the sniff and parse. @Voxstrasza @TheLegendaryKingdom That info is per official sniffing and parsing, i wrote the sniffing standard guides so will let you two debate from there, your current adjustments is not too far from official values, but since i provided the official values, please reconsider readjusting your values to as close as possible to the official values.

In my opinion, use whatever looks better.  We know he's supposed to be on the platform, we know his feet and weapon shouldn't clip with the platform - as long as the coords are within reason I don't think exact measurements matter for this one

**Comment by Voxstrasza** [comment]
> > @Nyeriah i submitted the sniff via the sniff team repo for your own personal comparison if you want. But at this point, it is up for debate, although the parsing shows a bit of a slight adjustment, it isnt much but may be much depending on how the Azerothcore is wanting to accept whatever adjustments based on the sniff and parse. @Voxstrasza @TheLegendaryKingdom That info is per official sniffing and parsing, i wrote the sniffing standard guides so will let you two debate from there, your current adjustments is not too far from official values, but since i provided the official values, please reconsider readjusting your values to as close as possible to the official values.
> 
> I tested with the values your parsed, and **after the first restart LK's feet were into the ground**. However, the other coordinates seem more accurate. So, I only edited position_z from the parse to be **161.21** and after a dozen more restarts it did **not** merge to the ground.
> 
> By the way, I used the raw values and like @Kitzunu said _it automatically rounded_ them.
> 
> I suggest using the **official values** and adjust only `position_z = 161.21`.

Since you have it loaded, can you post side-by-side screenshots?

**Comment by Kitzunu** [comment]
I think we go off raw sniff values. Because they are the correct ones
Perhaps the gob is misplaced and not the creature :)

**Comment by Kitzunu** [comment]
Can someone get me the gob entry? Im not able to go on the PC today

**Comment by Nyeriah** [comment]
The problem is that temporary fixes become permanent fixes rather frequently, and changing values away from the parsed ones will often overshadow the actual issue - creatures might be icorrectly spawned when they're placed on top of gameobjects. If we use the value from retail and it's still cliping or showing icorrectly ingame the issue is not his spawn, but rather the way the core is spawning him.

**Comment by Voxstrasza** [comment]
> The problem is that temporary fixes become permanent fixes rather frequently, and changing values away from the parsed ones will often overshadow the actual issue - creatures might be incorrectly spawned when they're placed on top of gameobjects. If we use the value from retail and it's still cliping or showing incorrectly ingame the issue is not his spawn, but rather the way the core is spawning him.

Personally, I'd rather adjust the xyz and revisit the xyz once it's figured out why the clipping is happening.  It's better than having an entire model fall into a gob.  Your call.

**Comment by Voxstrasza** [comment]
However, sniffs shouldn't be taken as face value.  They're sometimes incorrect.  They should be used as a base.

**Comment by acidmanifesto** [comment]
> However, sniffs shouldn't be taken as face value. They're sometimes incorrect. They should be used as a base.

Not sniffs being incorrect. It is the parsing that will provide diffrerent accuracies over time with various improvements. But the Sniff Packets will always remain the same as it is a raw capture of the complete session of all packet movement\exchanges\acknowledges.

**Comment by Voxstrasza** [comment]
> Can someone get me the gob entry? Im not able to go on the PC today

I believe the platform is made up of two gobs that are 191203 and 191204

**Comment by acidmanifesto** [comment]
Currently the only inaccuracies or inability to parse with the DK Starterzone as a Human DK is Hotfixes which are not relevant for anything before cata expansion where cata is when hotfixes were first introduced. Running it over with SL to check level squashing adjustments also shows the same xyzo as posted earlier.

image

**Comment by TheLegendaryKingdom** [comment]
> Since you have it loaded, can you post side-by-side screenshots?

With the value 161.21 in position_z

WoWScrnShot_120421_223211
WoWScrnShot_120421_223029
WoWScrnShot_120421_223023
WoWScrnShot_120421_222932
WoWScrnShot_120421_222919
WoWScrnShot_120421_222917

**Comment by Voxstrasza** [comment]
@Nyeriah @acidmanifesto @Kitzunu I'm going to shelve this PR for now until more info is gathered. Either the sniffed coordinates are incorrect, the gob is incorrect or the server doesn't spawn models in correctly.

**Comment by acidmanifesto** [comment]
> @Nyeriah @acidmanifesto @Kitzunu I'm going to shelve this PR for now until more info is gathered. Either the sniffed coordinates are incorrect, the gob is incorrect or the server doesn't spawn models in correctly.

Personally. I say go off what i provided for the z axis. Ours is off slightly.

**Comment by Voxstrasza** [comment]
> > @Nyeriah @acidmanifesto @Kitzunu I'm going to shelve this PR for now until more info is gathered. Either the sniffed coordinates are incorrect, the gob is incorrect or the server doesn't spawn models in correctly.
> 
> Personally. I say go off what i provided for the z axis. Ours is off slightly.

Applied to PR

**Comment by TheLegendaryKingdom** [comment]
> > @Nyeriah @acidmanifesto @Kitzunu I'm going to shelve this PR for now until more info is gathered. Either the sniffed coordinates are incorrect, the gob is incorrect or the server doesn't spawn models in correctly.
> 
> Personally. I say go off what i provided for the z axis. Ours is off slightly.

I'm confused, did you mean to use the parsed coords except for the z axis ?

**Comment by acidmanifesto** [comment]
> > > @Nyeriah @acidmanifesto @Kitzunu I'm going to shelve this PR for now until more info is gathered. Either the sniffed coordinates are incorrect, the gob is incorrect or the server doesn't spawn models in correctly.
> > 
> > Personally. I say go off what i provided for the z axis. Ours is off slightly.
> 
> I'm confused, did you mean to use the parsed coords except for the z axis ?

No. I am saying used the z-axis as there is supporting information with a sniff coordinate.

**Comment by Voxstrasza** [comment]
@TheLegendaryKingdom can you test 2310.27, -5742.21, 161.207, 3.85718?

**Comment by TheLegendaryKingdom** [comment]
> 161.207

Let me a few minutes, I do this (:

**Comment by TheLegendaryKingdom** [comment]
Well, the dozen of restarts has been done. No problem so far
WoWScrnShot_120521_031555
WoWScrnShot_120521_031528
.

**Comment by TheLegendaryKingdom** [comment]
With some details eventuall
WoWScrnShot_120521_031719
y

**Comment by Voxstrasza** [comment]
Looks good to me.

**Comment by TheLegendaryKingdom** [comment]
> Looks good to me.

I agree ! I dunno if he might still unfortunately and randomly clip in the ground, but it seems good like this.
I just remember the first time I tried these parsed coordinates, LK was clipping (maybe I had wrote them bad) (x

**Comment by TheLegendaryKingdom** [comment]
**OH NO ! He did it again !**
WoWScrnShot_120521_033442

**Comment by Voxstrasza** [comment]
well that stinks

**Comment by Voxstrasza** [comment]
2310.64, -5742.08, 161.3, 3.85123 <- this is the coordinate that works
2310.27, -5742.21, 161.207, 3.85718 <- this is the sniff coordinate that doesnt

**Comment by ayahuasca-ruderalis** [comment]
> 2310.64, -5742.08, 161.3, 3.85123 <- this is the coordinate that works 2310.27, -5742.21, 161.207, 3.85718 <- this is the sniff coordinate that doesnt

To be closer, **161.21** worked well too.

**Comment by TheLegendaryKingdom** [comment]
> 2310.64, -5742.08, 161.3, 3.85123 <- this is the coordinate that works 2310.27, -5742.21, 161.207, 3.85718 <- this is the sniff coordinate that doesnt

To be closer, **161.21** worked well too.

At least I did not experience any clip in my restarts.

**Comment by TheLegendaryKingdom** [comment]
I mean, it's only the position_z that make it clip imo. The other from the sniff should be good.
I used `x: 2310.27 y: -5742.21, z: 161.21 o: 3.85718`

**Comment by Voxstrasza** [comment]
I am setting ```2310.27, -5742.21, 161.21``` as the coordinates for this PR.  WIP can be removed.

File: data/sql/updates/pending_db_world/rev_1638646536763994700.sql
Comment (TheLegendaryKingdom) [inline data/sql/updates/pending_db_world/rev_1638646536763994700.sql:8]: Shouldn't add a new line at EOF ?

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1638646536763994700.sql:8]: there is

**Review by TheLegendaryKingdom — APPROVED** [review]

**Comment by TheLegendaryKingdom** [comment]
@Voxstrasza While in the bus going to work, I was completing DK starting quests in order to make a list of all of them. I do this in order to reward all DK starting quests within the skip-dk module. As I came to the "Lich King on his platform" last phase, using the coords we said, LK was felt INSIDE the platform. Not his feet getting merged to the platform's ground this time, the base problem occured again with the coords I thought fixes it :/

**Comment by acidmanifesto** [comment]
> @Voxstrasza While in the bus going to work, I was completing DK starting quests in order to make a list of all of them. I do this in order to reward all DK starting quests within the skip-dk module. As I came to the "Lich King on his platform" last phase, using the coords we said, LK was felt INSIDE the platform. Not his feet getting merged to the platform's ground this time, the base problem occured again with the coords I thought fixes it :/

This seems to be a sporadic issue still even with tc. Although the occurrence is greatly reduced. Seems to be a bit persistent from tc reports. I may need to see about finding a alternative way of handling the issue and will explore a few other options. 
I do have a couple of questions.
When encountering the issue do u have any custom adjustments dealing with visibility and draw distance outside of the default conf?
Is this occurring during natural quest or with gm mode on?

**Comment by TheLegendaryKingdom** [comment]
@acidmanifesto This error did not occur the other day when I tested as GM. This time in the bus I was regularly playing the questline.

I do not have any custom adjustments dealing with visibility and draw distance.

I also already experienced this issue on TC before coming to AC (less than a few months ago).

After I restarted the server LK was back on top of his platform, acting like nothing happenned (what a scammer !).

**Comment by acidmanifesto** [comment]
I have a possible hack fix to correct this but i will get with the az core staff first before i make any pr about it.
image

**Comment by TheLegendaryKingdom** [comment]
> I have a possible hack fix to correct this but i will get with the az core staff first before i make any pr about it. image

Nice to hear. It seems legit making it checked by staff as it is hacky way of fixing things. Did you find out a bit of what causes the issue ?

**Comment by acidmanifesto** [comment]
> > I have a possible hack fix to correct this but i will get with the az core staff first before i make any pr about it. image
> 
> Nice to hear. It seems legit making it checked by staff as it is hacky way of fixing things. Did you find out a bit of what causes the issue ?

the solution is actually found in a upcoming pr we are working on waiting to be merged, which is the inhabitremoval pr: https://github.com/azerothcore/azerothcore-wotlk/pull/9272

I have the individual adjustment for that lich king pending to be pushed for that pr to correct it, however i am waiting for feed back in the staff room about it

**Comment by Voxstrasza** [comment]
> > > I have a possible hack fix to correct this but i will get with the az core staff first before i make any pr about it. image
> > 
> > 
> > Nice to hear. It seems legit making it checked by staff as it is hacky way of fixing things. Did you find out a bit of what causes the issue ?
> 
> the solution is actually found in a upcoming pr we are working on waiting to be merged, which is the inhabitremoval pr: #9272
> 
> I have the individual adjustment for that lich king pending to be pushed for that pr to correct it, however i am waiting for feed back in the staff room about it

@acidmanifesto what should I do with this PR?

**Comment by acidmanifesto** [comment]
Tbh. It can still be applicable with the position adjustment even when the other pr passes.

**Comment by acidmanifesto** [comment]
Closes due to fix: https://github.com/azerothcore/azerothcore-wotlk/pull/9272

**Pull request closed without merging**
